benchmarks/microbenchmarks/_plan.py

```diff
@@ -49,10 +49,13 @@ def _get_kwargs(self, tags):
 
     def _init_taichi(self, arch, tags):
         for tag in tags:
-            if ResultType.init_taichi(arch, tag):
+            if MetricType.init_taichi(arch, tag):
                 return True
         return False
 
     def _run_func(self, tags: list):
-        return self.func(self.arch, self.basic_repeat_times,
-                         **self._get_kwargs(tags))
+        for tag in tags:
```

Comment:
Maybe we can generate `kwargs` in function `run`, and then define a item `func_item` to find the function in `func_lut`, then we can get the function by `self.func_lut[kwargs[func_item]]`?